converters/fromZigbee.js

```diff
@@ -3939,6 +3939,42 @@ const converters = {
                         {sunday_p3: value[30] + 'h:' + value[31] + 'm ' + value[32]/2 + '°C'},
                         {sunday_p4: value[33] + 'h:' + value[34] + 'm ' + value[35]/2 + '°C'},
                     ],
+                    weekdays_p1_hour: value[0],
```

Comment:
Yes, it is a replacement, but I was unsure sure whether removing `program` would be a breaking change for some people. However, if `program` is left as it is without all these new fields, I'm not sure how they can be sensibly exposed to the user. Do you have any recommendations?

Reply:
Can you check how well configuring of the program works via the z2m frontend -> device -> exposes tab? With this change a lot of extra sensors are discovered to HA which I don't think is a good idea.

Reply:
I've checked that, it shows 36 sliders in z2m frontend and they all work as expected (including the 0.5C increment for temperature).

I did also try exposing this as a `composite` item, which works well in the z2m frontend (although still shows 36 sliders, they are just grouped together with an apply button at the bottom). However the composite doesn't get exposed to HA.

What would be the alternative to exposing 36 sensors? The only alternative I can think of is making it a single `exposes.text()` with the 36 concatenated numbers, as some other thermostats do, but we then need to parse and validate the string, which also isn't as nice for the user to have to get the string format right in order for it to be accepted.

In the end there are 36 configurable values so does it not make sense to expose all 36? They could be exposed as configuration values rather than device controls, bit that would require another change to z2m as well.

Reply:
My preference would be to leave it as is. So you can configure the program through the z2m frontend (where we have more flexibility) and not HA.

Reply:
Currently the program can't be configured at all, even through the z2m frontend. It only gets shown in the state, but it's currently not included in the `exposes` tab at all. So even if we ignore HA there is still no way to set the program from z2m. So if we leave the `program` state as it is, in order to configure it from the z2m frontend we would use 12 `exposes.text()` fields and then parse these text fields to get back to the integers needed to send to the device. Would that be your preference over 36 sliders?

Reply:
> I did also try exposing this as a composite item, which works well in the z2m frontend (although still shows 36 sliders, they are just grouped together with an apply button at the bottom).

This is the best solution IMO, can you update the PR with this?

Reply:
I can update it with this (composite with 36 sliders) but that still requires a change to the `state` to have the 36 numbers as individual fields rather than as concatenated text like they currently are. The current list of objects format for `program` in the state just doesn't fit in with any of the ways of exposing it, although I misunderstand how the `expose` functions work. I think I'd need to change the format to this to be compatible with a `composite` expose of numeric format:

```js
{
    "program": {
        "weekdays_p1_hour": value[0],
        "weekdays_p1_minute": value[1],
        "weekdays_p1_temperature": value[2] / 2,
        ...
        "sunday_p4_temperature": value[35] / 2
    }
}
```

Is it ok to change the format or will it break things for other people? I see that this format was previously changed in #2334 so I guess it's ok to do this?

Reply:
Now I'm trying this and as I said previously this does make it impossible to set the thermostat program from HA. If the philosophy of z2m is to work with the z2m frontend and not worry about HA then fine, but personally if we do it like this then this PR doesn't help me and I'm going to have to use a custom converter similar to that in my original PR in order to make this appear in HA, as will anybody else who wants to do the same. If there are going to be 36 sliders anyway in the z2m front end, would it not make sense to put them as separate `numeric` exposes rather than a `composite` for the sake of it being easy to get to from HA? Or is there a way of integrating with HA which I'm missing?